ibattery.py

```diff
@@ -0,0 +1,237 @@
+
+    def configure_(self, percent, plugged):
+        if plugged:
+            self.foreground = self.charging_fg
+        elif percent <= self.warn_below:
+            self.foreground = self.low_foreground
+        else:
+            self.foreground = self._foreground
+
+    def calculate_length(self):
+        if self.bar.horizontal:
+            return (self.padding * 2) + self.widget_width
+        else:
+            return 0
+
+    def draw(self):
+        percent, charging = self.get_bat()
+        self.configure_(percent, charging)
+        self.draw_battery(percent, charging)
+
+    def draw_battery(self, percent, charging):
+        self.drawer.clear(self.background or self.bar.background)
+        if self.bar.horizontal:
+            PERCENT = self.BAR_WIDTH / 100 * percent
+            y_margin = (self.bar.height - self.HEIGHT) / 2
+
+            self.drawer.set_source_rgb("8c8c8c")
+            self._fill_body(
+                1,
+                y_margin,
+                width=self.BAR_WIDTH,
+                height=self.HEIGHT,
+                linewidth=1
+            )
+            self.drawer.set_source_rgb(self.foreground)
+            self._border(
+                1,
+                y_margin,
+                width=self.BAR_WIDTH,
+                height=self.HEIGHT,
+                linewidth=2.6
+            )
+            if percent <= self.warn_below:
+                self.drawer.set_source_rgb(self.low_foreground)
+            else:
+                self.drawer.set_source_rgb(self.foreground if charging else "ff8c1a")
+            self._fill_body(
+                2,
+                y_margin,
+                width=PERCENT,
+                height=self.HEIGHT,
+                linewidth=1
+            )
+            self.drawer.set_source_rgb("000000")
+            self._border(
+                1,
+                y_margin,
+                width=self.BAR_WIDTH,
+                height=self.HEIGHT,
+                linewidth=0.6
+            )
+            self.drawer.set_source_rgb(self.foreground)
+            self._fill_body(
+                self.BAR_WIDTH - 2,
+                y_margin + 1,
+                width=8.3,
+                height=self.HEIGHT - 2,
+                linewidth=5
+            )
+            self.drawer.ctx.select_font_face(
+                "sans", cairo.FONT_SLANT_NORMAL, cairo.FONT_WEIGHT_BOLD
+            )
+            self.drawer.ctx.set_font_size(12)
+            self.drawer.ctx.move_to(5, self.HEIGHT)
+            self.drawer.set_source_rgb("ffffff")
+            self.drawer.ctx.show_text(str(percent))
+
+            self.drawer.draw(
+                offsetx=self.offset + self.padding,
+                offsety=self.offsety,
+                width=self.length
+            )
+
+    def get_bat(self):
+        battery = psutil.sensors_battery()
+        try:
+            plugged = battery.power_plugged
+        except AttributeError:
+            return "No battery found."
```

Comment:
Your method is expected to return a tuple of the percentage and its plugged state. Returning a single string will break the unpacking in line 110.